figures/STable2/scripts/STable.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pathlib
+
+import pandas as pd
+import toml
+
+# In[2]:
+
+
+# set the path to the data
+path = pathlib.Path(
+    "../../../1.Exploratory_Data_Analysis/results/PBMC_combined.parquet"
+)
+
+# load in the data
+df = pd.read_parquet(path)
+
+
+# In[3]:
+
+
+# path to the ground truth
+ground_truth_path = pathlib.Path(
+    "../../../4.sc_Morphology_Neural_Network_MLP_Model/MLP_utils/ground_truth.toml"
+)
+# load in the ground truth
+ground_truth = toml.load(ground_truth_path)
+apoptosis_ground_truth = ground_truth["Apoptosis"]["apoptosis_groups_list"]
+pyroptosis_ground_truth = ground_truth["Pyroptosis"]["pyroptosis_groups_list"]
+control_ground_truth = ground_truth["Healthy"]["healthy_groups_list"]
+
+
+# In[4]:
+
+
+# change the p-adj into absolute values
+df["p-adj"] = df["p-adj"].abs()
+df.head()
+# select row that have p-adj < 0.05
+df = df[df["p-adj"] < 0.05]
+
+
+# In[5]:
+
+
+# add the group1 and group2 columns into 1 column
+df["group"] = df["group1"] + "_" + df["group2"]
+
+
+# In[6]:
+
+
+# set theory for apoptosis control and pyroptosis
+apoptosis_vs_healthy = df[df["group"] == "apoptosis_healthy"]
+pyroptosis_vs_healthy = df[df["group"] == "healthy_pyroptosis"]
+pyroptosis_vs_apoptosis = df[df["group"] == "apoptosis_pyroptosis"]
+
+# get thee list of genes that are significant from each comparision
+A = apoptosis_vs_healthy["features"].tolist()  # apoptosis_vs_healthy_list
+B = pyroptosis_vs_apoptosis["features"].tolist()  # pyroptosis_vs_apoptosis_list
+C = pyroptosis_vs_healthy["features"].tolist()  # pyroptosis_vs_healthy_list
```

Comment:
Consider adding additional description to the variable names here, perhaps something like `set_A`, etc.